addons/onedrive/client.py

```diff
@@ -54,10 +59,54 @@ def user_info_for_token(self, access_token):
         :rtype: dict
         :return: a dict containing metadata about the token's owner.
         """
-        return self._make_request(
+
+        # get user properties from /me endpoint
+        me_url = self._build_url(settings.MSGRAPH_API_URL, 'me')
+        me_resp = self._make_request(
             'GET',
-            self._build_url(settings.MSLIVE_API_URL, 'me'),
-            params={'access_token': access_token},
+            me_url,
+            headers=self._default_headers,
             expects=(200, ),
             throws=HTTPError(401)
-        ).json()
+        )
+        me_data = me_resp.json()
+        logger.debug('me_data:({})'.format(me_data))
+
+        retval = {
+            'id': me_data['id'],
+            'name': me_data['displayName'],
+            'link': '{}/users/{}'.format(settings.MSGRAPH_API_URL, me_data['id']),
+            'mail': me_data['userPrincipalName'],
+        }
+
+        # get drive properties from /users/$user_id/drive endpoint
+        drive_url = self._build_url(settings.MSGRAPH_API_URL, 'users', retval['id'], 'drive')
+        drive_resp = self._make_request(
+            'GET',
+            drive_url,
+            headers=self._default_headers,
+            expects=(200, ),
+            throws=HTTPError(401)
+        )
+        drive_data = drive_resp.json()
+        logger.debug('drive_data:({})'.format(drive_data))
+        retval['drive_id'] = drive_data['id']
+
+        if drive_data['driveType'] == 'personal':
+            retval['name'] = '{} - OneDrive Personal'.format(retval['mail'])
+        else:
+            # get site properties from /sites endpoint
+            # site_url = self._build_url(settings.MSGRAPH_API_URL, 'sites', 'root')
+            # site_resp = self._make_request(
+            #     'GET',
+            #     site_url,
+            #     headers=self._default_headers,
+            #     expects=(200, ),
+            #     throws=HTTPError(401)
+            # )
+            # site_data = site_resp.json()
+            # logger.debug('site_data:({})'.format(site_data))
```

Comment:
Is this request to `\sites` endpoint for local debugging only?

Reply:
This is related to the section of the commit message titled "feature: use email + account type for user setting name". I was trying to find a way to uniquely label a user+site connection.  I hoped that the site metadata might have something I could use.  For my available test cases, it didn't, but I've left in the commented-out code in b/c it *may* be useful again in the future.